heron/common/src/cpp/network/client.cpp

```diff
@@ -181,8 +181,10 @@ void Client::OnNewPacket(IncomingPacket* _ipkt) {
 
   if (_ipkt->UnPackString(&typname) != 0) {
     Connection* conn = static_cast<Connection*>(conn_);
-    LOG(FATAL) << "UnPackString failed from connection " << conn << " from hostport "
+    LOG(ERROR) << "UnPackString failed from connection " << conn << " from hostport "
```

Comment:
I am still not convinced by the idea of pull request. This method returns nothing. So this failure will be no way visible to higher layer's logic: we can not handle this dropping. 
I would still prefer the fail-fast.
Could you describe more on the scenarios of corruption issue?